frontend/app/src/components/section-components/create-channel-form.tsx

```diff
@@ -0,0 +1,146 @@
+import { Dispatch, useState } from "react";
+import { channelType } from "../global-components/interface";
+import useCreateChannel from "../query-hooks/useCreateChannel";
+
+interface CreateChannelFormProps {
+  setShowForm: Dispatch<React.SetStateAction<boolean>>;
+}
+
+const defaultFormData = {
+  name: "",
+  type: channelType.Public,
+  password: "",
+}
+
+function CreateChannelForm(props: CreateChannelFormProps) {
+  const createChannelMutation = useCreateChannel();
+  const [formData, setFormData] = useState(defaultFormData);
+  const [passwordRequired, setPasswordRequired] = useState(false);
+  const [inputStatus, setInputStatus] = useState<string>('empty');
+  const { name, password } = formData;
+
+  function onChange(e: React.ChangeEvent<HTMLInputElement>) {
+    setInputStatus('editing');
+    setFormData((prevState) => ({
+      ...prevState,
+      [e.target.id]: e.target.value,
+    }));
+  }
+
+  function checkFormData(formData: {name: string, type: channelType, password: string}) {
```

Comment:
Could separate CheckFormData into 2 functions to avoid handling error codes and to speacialized functions
> isValidName(){} : boolean 
> isValidPassword(){}: boolean

Reply:
Should check for unauthorized characters such as spaces

Reply:
Maybe open a new issue and bundle password hashing + front checking of form fields and back checking of channel name